### PR TITLE
Add a codec parser in front of the pipeline

### PR DESCRIPTION
**Pull request by kekkokk**
Added a codec parser in front of pipeline to help keep track of codec and clock rate in entire pipeline for  each packet

[] It needs and includes Unit Tests
[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

File: erizo/src/erizo/MediaStream.cpp [hunk 1]
```diff
@@ -188,6 +189,7 @@ void MediaStream::initializePipeline() {
   pipeline_->addFront(SenderBandwidthEstimationHandler());
   pipeline_->addFront(LayerDetectorHandler());
   pipeline_->addFront(OutgoingStatsHandler());
+  pipeline_->addFront(PacketCodecParser());
```
Comment (jcague) [inline erizo/src/erizo/MediaStream.cpp]: 👍

File: erizo/src/erizo/rtp/LayerDetectorHandler.cpp [hunk 2]
```diff
@@ -34,16 +34,11 @@ void LayerDetectorHandler::disable() {
 void LayerDetectorHandler::read(Context *ctx, std::shared_ptr<DataPacket> packet) {
   RtcpHeader *chead = reinterpret_cast<RtcpHeader*>(packet->data);
   if (!chead->isRtcp() && enabled_ && packet->type == VIDEO_PACKET) {
-    RtpHeader *rtp_header = reinterpret_cast<RtpHeader*>(packet->data);
-    RtpMap *codec = stream_->getRemoteSdpInfo()->getCodecByExternalPayloadType(rtp_header->getPayloadType());
-    if (codec && codec->encoding_name == "VP8") {
-      packet->codec = "VP8";
+    if (packet->codec = "VP8") {
```
Comment (jcague) [inline erizo/src/erizo/rtp/LayerDetectorHandler.cpp]: `==` ?

File: erizo/src/erizo/rtp/LayerDetectorHandler.cpp [hunk 3]
```diff
@@ -34,16 +34,11 @@ void LayerDetectorHandler::disable() {
...
       parseLayerInfoFromVP8(packet);
-    } else if (codec && codec->encoding_name == "VP9") {
-      packet->codec = "VP9";
+    } else if (packet->codec = "VP9") {
```
Comment (jcague) [inline erizo/src/erizo/rtp/LayerDetectorHandler.cpp]: same as above

File: erizo/src/erizo/rtp/LayerDetectorHandler.cpp [hunk 4]
```diff
@@ -34,16 +34,11 @@ void LayerDetectorHandler::disable() {
...
       parseLayerInfoFromVP9(packet);
-    } else if (codec && codec->encoding_name == "H264") {
-      packet->codec = "H264";
+    } else if (packet->codec = "H264") {
```
Comment (jcague) [inline erizo/src/erizo/rtp/LayerDetectorHandler.cpp]: same as above

File: erizo/src/erizo/rtp/LayerDetectorHandler.cpp (hunk 2 above)
Comment (kekkokk) [reply erizo/src/erizo/rtp/LayerDetectorHandler.cpp]: yeah, I'm trying to do my best in recreating the prs from my own deviated source code. fixed typos

**Comment by jcague** [comment]
Please, take a look at CircleCI to fix the lint issues.

**Pull request merged**